lib/ansible/playbook/role/__init__.py

```diff
@@ -68,7 +68,7 @@ def hash_params(params):
                 new_params = set()
                 for k, v in params.items():
                     # Hash each entry individually
-                    new_params.update((k, hash_params(v)))
+                    new_params.add((k, hash_params(v)))
```

Comment:
won't this give different behaviour compared to `frozenset` above if `.items()` is non unique?